### PR TITLE
feat(pubsublite): add TopicStats service

### PR DESCRIPTION
**Pull request by coryan**
This service does not use any streaming RPCs, it is suitable for code
generation.

Part of the work for #7670

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7966)

### CODE REVIEW

File: google/cloud/pubsublite/CMakeLists.txt [hunk 1]
```diff
@@ -85,7 +85,28 @@ add_library(
...
+    internal/topic_stats_metadata_decorator.cc
+    internal/topic_stats_metadata_decorator.h
+    internal/topic_stats_option_defaults.cc
+    internal/topic_stats_option_defaults.h
+    internal/topic_stats_retry_traits.h
+    internal/topic_stats_stub.cc
+    internal/topic_stats_stub.h
+    internal/topic_stats_stub_factory.cc
+    internal/topic_stats_stub_factory.h
+    mocks/mock_topic_stats_connection.h
```
Comment (dbolduc) [inline google/cloud/pubsublite/CMakeLists.txt:102]: this should be in the mock library

File: google/cloud/pubsublite/google_cloud_cpp_pubsublite.bzl [hunk 2]
```diff
@@ -29,6 +29,18 @@ google_cloud_cpp_pubsublite_hdrs = [
...
     "internal/admin_stub_factory.h",
     "internal/stream_factory.h",
+    "internal/topic_stats_auth_decorator.h",
+    "internal/topic_stats_logging_decorator.h",
+    "internal/topic_stats_metadata_decorator.h",
+    "internal/topic_stats_option_defaults.h",
+    "internal/topic_stats_retry_traits.h",
+    "internal/topic_stats_stub.h",
+    "internal/topic_stats_stub_factory.h",
+    "mocks/mock_topic_stats_connection.h",
```
Comment (dbolduc) [inline google/cloud/pubsublite/google_cloud_cpp_pubsublite.bzl:39]: "

**Review by dbolduc — CHANGES_REQUESTED** [review]
Also, do we have a code coverage plan? Right now we ignore all of `google/cloud/pubsublite`. But if there is hand written code, we will want to test that.

A not so great option is to move this and the Admin API into a subdirectory, then ignore that subdirectory. Food for thought.

File: google/cloud/pubsublite/CMakeLists.txt (hunk 1 above)
Comment (coryan) [reply google/cloud/pubsublite/CMakeLists.txt]: Thanks for catching that, fixed.

**Comment by coryan** [comment]
> Also, do we have a code coverage plan? Right now we ignore all of `google/cloud/pubsublite`. But if there is hand written code, we will want to test that.

Eventually we will need more complicated rules for this directory, I think maybe using this:

https://docs.codecov.com/docs/path-regexp

**Review by dbolduc — APPROVED** [review]
codecov regex SGTM

**Pull request merged**